src/test/regress/sql/with.sql

```diff
@@ -1190,3 +1190,17 @@ create temp table with_test (i int);
 with with_test as (select 42) insert into with_test select * from with_test;
 select * from with_test;
 drop table with_test;
+
+-- Test that planner correctly assigns writer and reader slices in case of
```

Comment:
I run test and got such diff
```diff
diff -I HINT: -I CONTEXT: -I GP_IGNORE: -U3 /home/src/gpdb7/src/test/regress/expected/with_optimizer.out /home/src/gpdb7/src/test/regress/results/with.out
--- /home/src/gpdb7/src/test/regress/expected/with_optimizer.out    2023-12-27 14:47:50.070179849 +0500
+++ /home/src/gpdb7/src/test/regress/results/with.out   2023-12-27 14:47:50.114180633 +0500
@@ -2460,9 +2460,9 @@
                                              ->  Result
  Slice 0: Dispatcher; root 0; parent -1; gang size 0
  Slice 1: Reader; root 0; parent 0; gang size 3
- Slice 2: Reader; root 0; parent 1; gang size 3
- Slice 3: Primary Writer; root 0; parent 1; gang size 3
- Slice 4: Singleton Reader; root 0; parent 3; gang size 1; segment 1
+ Slice 2: Primary Writer; root 0; parent 1; gang size 3
+ Slice 3: Reader; root 0; parent 1; gang size 3
+ Slice 4: Singleton Reader; root 0; parent 3; gang size 1; segment 0
 GP_IGNORE:(21 rows)

 DROP TABLE withmodcte;
```

Reply:
Well, I see. It will be more reliable to test only execution in that case.
gpdb7 is upstream version? In my build I ran tests from adb-7.0.0 branch.
Moreover, there aren't any CI tests for our 7.x  branch, and I doubt that we will start merging tickets before CI support.

Reply:
> gpdb7 is upstream version?

No! I just run test on ADBDEV-4345 branch

Reply:
Why does the same (Postgres-based) planner produce different results?

Reply:
If I run the whole test suit from parallel_schedule (before the group with `with` test), all the tests pass correctly.

Reply:
I run so:
```sh
./pg_regress --load-extension=gp_inject_fault --init-file=init_file create_function_1 create_type create_table int8 int4 copy with
```

Reply:
hmm...
```diff
diff -I HINT: -I CONTEXT: -I GP_IGNORE: -U3 /home/src/gpdb7/src/test/regress/expected/with.out /home/src/gpdb7/src/test/regress/results/with.out
--- /home/src/gpdb7/src/test/regress/expected/with.out  2023-12-27 16:09:01.065718112 +0500
+++ /home/src/gpdb7/src/test/regress/results/with.out   2023-12-27 16:09:01.109718736 +0500
@@ -2460,9 +2464,9 @@
                                              ->  Result
  Slice 0: Dispatcher; root 0; parent -1; gang size 0
  Slice 1: Reader; root 0; parent 0; gang size 3
- Slice 2: Reader; root 0; parent 1; gang size 3
- Slice 3: Primary Writer; root 0; parent 1; gang size 3
- Slice 4: Singleton Reader; root 0; parent 3; gang size 1; segment 2
+ Slice 2: Primary Writer; root 0; parent 1; gang size 3
+ Slice 3: Reader; root 0; parent 1; gang size 3
+ Slice 4: Singleton Reader; root 0; parent 3; gang size 1; segment 1
 GP_IGNORE:(21 rows)

 DROP TABLE withmodcte;
```

Reply:
At least `segment 2/1`  part is flacky. It means that we need either add `matchsubs` , or just test the execution only

Reply:
what about order of `Slice 2: Primary Writer` and `Slice 3: Reader` ?

Reply:
I can't say anything without a plan.

Reply:
plans are same (as in expected files)

Reply:
The code is idepended from enviroment, I'm getting correct plan, and the plan creation procedure is not ambiguous. 
I have the following diff on the same command (flacky part) or don't have diff at all:
```diff
--- /home/bimbo/Downloads/test_repo/src/test/regress/expected/with_optimizer.out	2023-12-27 14:25:56.010010388 +0300
+++ /home/bimbo/Downloads/test_repo/src/test/regress/results/with.out	2023-12-27 14:25:56.040010458 +0300
@@ -2462,7 +2462,7 @@
  Slice 1: Reader; root 0; parent 0; gang size 3
  Slice 2: Reader; root 0; parent 1; gang size 3
  Slice 3: Primary Writer; root 0; parent 1; gang size 3
- Slice 4: Singleton Reader; root 0; parent 3; gang size 1; segment 1
+ Slice 4: Singleton Reader; root 0; parent 3; gang size 1; segment 2
 GP_IGNORE:(21 rows)
```